build(linux): add Ubuntu 25.04 support to linux_build.sh

activatekillswitch opened the pull request:
## Description
Updated the linux_build.sh file to include Ubuntu 25.04.
Set the cuda and gcc versions.

Tested and working.

Added the config for Ubuntu 25.04

Tested on my machine, so ymmv.

Alienware Area-51 R5
Intel i9-9820X
NVIDIA GeForce RTX 2080
Ubuntu 25.04
Gnome 48
Wayland
Kernel - Linux 6.14.0-15-generic

Note that I had to patch the cuda math headers as per [nvidia forum](https://forums.developer.nvidia.com/t/error-exception-specification-is-incompatible-for-cospi-sinpi-cospif-sinpif-with-glibc-2-41/323591)

## Screenshot

## Issues Fixed or Closed
- Resolves https://github.com/LizardByte/Sunshine/issues/3892

## Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Dependency update (updates to dependencies)
- [ ] Documentation update (changes to documentation)
- [ ] Repository update (changes to repository files, e.g. `.github/...`)

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated the in code docstring/documentation-blocks for new or existing methods/components

ReenigneArcher commented on scripts/linux_build.sh:
```diff
@@ -564,6 +564,15 @@ elif grep -q "Ubuntu 24.04" /etc/os-release; then
+  version="25.04"
+  package_update_command="${sudo_cmd} apt-get update"
+  package_install_command="${sudo_cmd} apt-get install -y"
+  cuda_version="12.9.0"
+  cuda_build="575.51.03"
+  gcc_version="12"
```
```suggestion
  cuda_version="11.8.0"
  cuda_build="520.61.05"
  gcc_version="11"
```

Can you try with these parameters? This gives better compatibility with older cards.

activatekillswitch commented:
I've updated to use the suggested builds.

I had a number of issues to resolve in the environment.

Output from compiler:

_update-alternatives: using /usr/bin/gcc-12 to provide /usr/bin/gcc (gcc) in auto mode

/sunshine/build/cuda/bin/../targets/x86_64-linux/include/crt/host_config.h:132:2:
  error: #error -- unsupported GNU version! gcc versions later than 11 are
  not supported! The nvcc flag '-allow-unsupported-compiler' can be used to
  override this version check; however, using an unsupported host compiler
  may cause compilation failure or incorrect run time execution.  Use at your
  own risk._

Updated to force gcc-11

_sudo update-alternatives --config gcc
There are 3 choices for the alternative gcc (providing /usr/bin/gcc).

  Selection    Path             Priority   Status
------------------------------------------------------------
* 0            /usr/bin/gcc-12   100       auto mode
  1            /usr/bin/gcc-11   100       manual mode
  2            /usr/bin/gcc-12   100       manual mode
  3            /usr/bin/gcc-14   100       manual mode

Press <enter> to keep the current choice[*], or type selection number: 1
update-alternatives: using /usr/bin/gcc-11 to provide /usr/bin/gcc (gcc) in manual mode_

Then received the same error on compilation about the header mismatch.

_/usr/include/x86_64-linux-gnu/bits/mathcalls.h(79): error: exception
  specification is incompatible with that of previous function "cospi"_

Updated the header to add "noexcept(true)" for the sin and cos functions.

Then built fine and worked.
